itests/hive-unit/src/test/java/org/apache/hadoop/hive/metastore/MetastoreTaskThreadTestImpl1.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hive.metastore;
+
+import org.apache.hadoop.conf.Configuration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ * An implementation of MetastoreTaskThread for testing metastore leader config.
+ */
+public class MetastoreTaskThreadTestImpl1 implements MetastoreTaskThread {
```

Comment:
looks like lots of duplicate code. Can task name be set dynamically to get rid of redundant codes ?

Reply:
It's not about TASK_NAME really. That's just an indicator. What we really need to test is loading a class and creating a thread when it's specified in config. For that we need different loadable classes. We could have created a base class and extended it but that adds one more class and there's nothing much value in de-duplicating the code. We do not duplicate the code to avoid hazards of duplicating the code like bug fixes, changes at multiple places. Nothing of that sort is possible here. Further, having the code duplicated gives future flexibility to change code in each of those classes like changing runFrequency() or actual run() implementation, which is more desirable.

Reply:
Changed names of the classes as per discussion.